packages/common/src/state/simple.ts

```diff
@@ -0,0 +1,185 @@
+    this.topA().set(address.toString(), undefined)
+  }
+
+  async modifyAccountFields(address: Address, accountFields: AccountFields): Promise<void> {
+    let account = await this.getAccount(address)
+    if (!account) {
+      account = new Account()
+    }
+    account.nonce = accountFields.nonce ?? account.nonce
+    account.balance = accountFields.balance ?? account.balance
+    account.storageRoot = accountFields.storageRoot ?? account.storageRoot
+    account.codeHash = accountFields.codeHash ?? account.codeHash
+    await this.putAccount(address, account)
+  }
+
+  async getContractCode(address: Address): Promise<Uint8Array> {
+    return this.topC().get(address.toString()) ?? new Uint8Array(0)
+  }
+
+  async putContractCode(address: Address, value: Uint8Array): Promise<void> {
+    this.topC().set(address.toString(), value)
+    if ((await this.getAccount(address)) === undefined) {
+      await this.putAccount(address, new Account())
+    }
+    await this.modifyAccountFields(address, { codeHash: keccak256(value) })
+  }
+
+  async getContractCodeSize(address: Address): Promise<number> {
+    const contractCode = await this.getContractCode(address)
+    return contractCode.length
+  }
+
+  async getContractStorage(address: Address, key: Uint8Array): Promise<Uint8Array> {
+    return this.topS().get(`${address.toString()}_${bytesToHex(key)}`) ?? new Uint8Array(0)
+  }
+
+  async putContractStorage(address: Address, key: Uint8Array, value: Uint8Array): Promise<void> {
+    this.topS().set(`${address.toString()}_${bytesToHex(key)}`, value)
+  }
+
+  async checkpoint(): Promise<void> {
+    this.add()
+  }
+  async commit(): Promise<void> {
+    const topA = this.accountStack.pop()
+    const topC = this.codeStack.pop()
+    const topS = this.storageStack.pop()
+    this.accountStack.pop()
+    this.codeStack.pop()
+    this.storageStack.pop()
+    this.accountStack.push(topA!)
+    this.codeStack.push(topC!)
+    this.storageStack.push(topS!)
+  }
+
+  async revert(): Promise<void> {
+    this.accountStack.pop()
+    this.codeStack.pop()
+    this.storageStack.pop()
+  }
+
+  async flush(): Promise<void> {}
+  clearCaches(): void {}
+
+  shallowCopy(): EVMStateManagerInterface {
+    const copy = new SimpleStateManager()
+    for (let i = 0; i < this.accountStack.length; i++) {
+      copy.accountStack.push(new Map(this.accountStack[i]))
+      copy.codeStack.push(new Map(this.codeStack[i]))
+      copy.storageStack.push(new Map(this.storageStack[i]))
+    }
+    return copy
+  }
+
+  // State root functionality not implemented
+  getStateRoot(): Promise<Uint8Array> {
+    throw new Error('Method not implemented.')
+  }
+  setStateRoot(): Promise<void> {
+    throw new Error('Method not implemented.')
+  }
+  hasStateRoot(): Promise<boolean> {
+    throw new Error('Method not implemented.')
+  }
+
+  // Only goes for long term create situations, skip
+  async clearContractStorage(): Promise<void> {}
```

Comment:
Should also throw.

I just glanced in EVM, I see one call to this method, but I do not think this would ever actually "clear" any storage (however there might be some exotic cases which I have not thought of)

Reply:
Have answered above.

I would think that reasoning about what `clearContractStorage()` is doing when called from within the `EVM` and eventually do changes on the EVM side is outside the scope of this PR.